Content/GUI/PassiveTree.cs

```diff
@@ -201,91 +216,156 @@ public override void Draw(SpriteBatch spriteBatch)
+		{
+			yAbove = MathF.Max(yAbove, 0);
+			yBelow = MathF.Min(yBelow, 0);
+		}
+		else
+		{
+			yAbove = MathF.Min(yAbove, 0);
+			yBelow = MathF.Max(yBelow, 0);
+		}
 
-			adjust += new Vector2(xAbove, yAbove);
-			adjust += new Vector2(xBelow, yBelow);
+		if (rec.Width < MathF.Abs(UITree.BotRightTree.X) + MathF.Abs(UITree.TopLeftTree.X))
+		{
+			xAbove = MathF.Max(xAbove, 0);
+			xBelow = MathF.Min(xBelow, 0);
+		}
+		else
+		{
+			xAbove = MathF.Min(xAbove, 0);
+			xBelow = MathF.Max(xBelow, 0);
+		}
 
-			_start += adjust;
+		adjust += new Vector2(xAbove, yAbove);
+		adjust += new Vector2(xBelow, yBelow);
 
-			foreach (UIElement element in Elements)
-			{
-				if (element is PassiveElement ele)
-				{
-					element.Left.Set(ele.Root.X + Main.MouseScreen.X - _start.X, 0.5f);
-					element.Top.Set(ele.Root.Y + Main.MouseScreen.Y - _start.Y, 0.5f);
-				}
-			}
+		offsetChange -= adjust;
 
-			_lineOff = _root + Main.MouseScreen - _start;
-		}
-		else
+		foreach (UIElement element in Elements)
 		{
-			_start = Vector2.Zero;
+			if (element is PassiveElement ele)
+			{
+				element.Left.Set(ele.Left.Pixels + offsetChange.X, 0.5f);
+				element.Top.Set(ele.Top.Pixels + offsetChange.Y, 0.5f);
+			}
 		}
 
+		_lineOff += offsetChange;
+
 		Main.blockMouse = _blockMouse;
 		_lastState = Main.mouseLeft;
 
 		Recalculate();
 	}
+}
+
+public class BlockClickItem : ModSystem
+{
+	// TODO: This is not blocking anything on the right. | or the open achivments button.
+
+	public static bool Block = false;
+	public override void Load()
+	{
+		On_ItemSlot.OverrideLeftClick += OverrideLeftClick_On;
+		On_ItemSlot.MouseHover_ItemArray_int_int += MouseHover_On;
+		On_UIElement.MouseOver += MouseOver_On;
+
+		PropertyInfo propertyInfo = typeof(PlayerInput).GetProperty("IgnoreMouseInterface",
+						 BindingFlags.Public |
+						 BindingFlags.Static);
+		MethodInfo methodInfo = propertyInfo.GetGetMethod();
+
+		MonoModHooks.Add(methodInfo, hook);
+	}
+
+	public override void ModifyInterfaceLayers(List<GameInterfaceLayer> layers)
+	{
+		Block = false;
+	}
+
+	private bool hook(Func<bool> action)
+	{
+		if (!Block)
+		{
+			return action();
+		}
+
+		return true;
+	}
+	public override void Unload()
+	{
+		On_ItemSlot.OverrideLeftClick -= OverrideLeftClick_On;
+		On_ItemSlot.MouseHover_ItemArray_int_int -= MouseHover_On;
+		On_UIElement.MouseOver -= MouseOver_On;
+	}
+
+	private void MouseOver_On(On_UIElement.orig_MouseOver orig, UIElement self, UIMouseEvent evt)
+	{
+		if (!Block)
+		{
+			orig(self, evt);
+		}
+	}
+
+	private bool OverrideLeftClick_On(On_ItemSlot.orig_OverrideLeftClick orig, Item[] inv, int context, int slot)
+	{
+		if (!Block)
+		{
+			return orig(inv, context, slot);
+		}
+
+		return true;
+	}
+
+	private void MouseHover_On(On_ItemSlot.orig_MouseHover_ItemArray_int_int orig, Item[] inv, int context, int slot)
+	{
+		if (!Block)
+		{
+			orig(inv, context, slot);
+		}
+	}
```

Comment:
Minor mod compat issue on all of these hooks, if another mod registers their hook before our mod then it will not be called unless we call `orig`. We can switch to using an IL edit, probably easiest to emit it all in IL, it should also be easy to reuse for each hook